### PR TITLE
bpo-30624 / selectors: use bare except clause

### PR DESCRIPTION
**Pull request by giampaolo**
References http://bugs.python.org/issue30624.

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]
Sounds good. Thanks! (Maybe we need an update to PEP 8 explaining that `except:` is okay when the block ends in bare `raise`?)

**Review by vstinner — APPROVED** [review]
LGTM. I don't merge immediately because I would like to know if other core devs like this change.

**Comment by gvanrossum** [comment]
Merging now so we can get this into the release candidate.

**Comment by gvanrossum** [comment]
Needs a backport to 3.6.

File: Misc/NEWS
Comment (serhiy-storchaka) [inline Misc/NEWS:353]: Not SystemExit.

**Review by serhiy-storchaka — COMMENTED** [review]
Look also at KqueueSelector.

File: Misc/NEWS
Comment (giampaolo) [reply Misc/NEWS:353]: Why not? `SystemExit` is not a subclass of `Exception`.

Comment (serhiy-storchaka) [reply Misc/NEWS:353]: You are right. My mistake.

**Comment by giampaolo** [comment]
> Look also at KqueueSelector.

What about it?

**Comment by serhiy-storchaka** [comment]
Sorry, I though you fixed first two occurrences of "except Exception", but actually the 1st and the 3rd are fixed.

**Comment by giampaolo** [comment]
> Needs a backport to 3.6.

I see that on 3.6 `except BaseException:` is used, so the backport should not be necessary.

**Comment by gvanrossum** [comment]
[For posterity, and note to myself] Oh, I see now what happened. Before #1035 this code was simpler and did have `except BaseException: / unregister() / raise` -- but @giampaolo changed that to `except Exception:` for some reason. It wasn't caught in review then (the PR makes it seem like a feature?) but I happened to mention it when reviewing #1030 (also by @giampaolo).

**Comment by zware** [comment]
@giampaolo There is still a `selectors-bare-except-3.6` branch in the repository that has no PR associated; does it contain something that still needs to be merged or can it be deleted?

**Comment by giampaolo** [comment]
It can be deleted.

**Pull request merged**
